ARKit Example/Classes/ARKit/View/RadarView.m

```diff
@@ -28,33 +28,55 @@ - (id)initAtPoint:(CGPoint)middlePoint {
     return self;
 }
 
-- (void) updatePoints:(ARGeoCoordinate *)centerCoord {
-	double centerAzimuth = centerCoord.azimuth + M_PI / 2;
-	if (centerAzimuth < 0.0) {
-		centerAzimuth = 2 * M_PI + centerAzimuth;
-	}
-	
-	UIImage *dot = [UIImage imageNamed:@"radar_dot.png"];
-	
-	for (UIView *sub in dotsView.subviews) {
-		[sub removeFromSuperview];
-	}
-	
-	for (ARGeoCoordinate *coord in points) {
-		double coordAzimuth = coord.azimuth - centerAzimuth;
-		if (coordAzimuth < 0.0) {
-			coordAzimuth = 2 * M_PI + coordAzimuth;
-		}
-		CGPoint pt;
-		
-		pt.x = dotsView.center.x + cos(coordAzimuth) * coord.radialDistance * self.frame.size.width / (2 * farthest);
-		pt.y = dotsView.center.y + sin(coordAzimuth) * coord.radialDistance * self.frame.size.height / (2 * farthest);
-		
-		UIImageView *point = [[UIImageView alloc] initWithImage:dot];
-		point.center = pt;
-		
-		[dotsView addSubview:point];
-	}
+- (void)updatePoints:(ARGeoCoordinate *)centerCoord reloadData:(BOOL)reloadData {
+    
+    double centerAzimuth = centerCoord.azimuth + M_PI / 2;
+    
+    if (centerAzimuth < 0.0f) {
+        centerAzimuth = 2 * M_PI + centerAzimuth;
+    }
+    
+    // Only remove subviews if reloading the data with new points
+    if (reloadData) {
```

Comment:
The point was for compatibility if you want to invoke reloaddata from superview without reallocating the entire radarview from beginning, for example you change positions and want to change the number of points.